packages/hive/src/Adaptor.js

```diff
@@ -0,0 +1,155 @@
+import {
+  execute as commonExecute,
+  composeNextState,
+} from '@openfn/language-common';
+import { expandReferences } from '@openfn/language-common/util';
+
+import hive from 'hive-driver';
+
+const { TCLIService, TCLIService_types } = hive.thrift;
+const utils = new hive.HiveUtils(TCLIService_types);
+
+const logger = message => console.log(message);
+
+/**
+ * Execute a sequence of operations.
+ * Wraps `language-common/execute` to make working with this API easier.
+ * @example
+ * execute(
+ *   create('foo'),
+ *   delete('bar')
+ * )(state)
+ * @private
+ * @param {Operations} operations - Operations to be performed.
+ * @returns {Operation}
+ */
+export function execute(...operations) {
+  const initialState = {
+    references: [],
+    data: null,
+  };
+
+  return state => {
+    return commonExecute(
+      connect,
+      ...operations,
+      cleanupState
+    )({
+      ...initialState,
+      ...state,
+    });
+  };
+}
+
+function connect(state) {
+  //TODO @Mtuchi how to use the databse?
```

Comment:
looks like this should be removed

Reply:
Done and added comment in improvement issue https://github.com/OpenFn/adaptors/issues/400#issuecomment-1735734535